### DISCUSSION TITLE
Recording/logging variables in a memory card.

### DISCUSSION THREAD

**Discussion by pberna67** [arendst/Tasmota#19854] [Q&A] [answered]
Dear Cmmunity

I would like to know if tasmota project includes the support of SD memory cards (with SPI or SD hw interface) and has the API support for storing variables in it . Could be interesting to store a variable with a related time stamp.

I would like to store a set temperatures with the time tag in the SD, periodically every 1s, of a network of 7 Dallas DS1820 1 wire tempertature sensors .  Is it possible?

Thanks

Regards
Paolo

**Comment by sfromis** [comment, marked as answer]
Yes, Tasmota can support a file system, if the feature is included in a build, including on a properly configured SD card. You can write files using [Berry](https://tasmota.github.io/docs/Berry/) programming on a ESP32 device. Or for ESP8266 using the [scripting](https://tasmota.github.io/docs/Scripting-Language/) feature (may take some time to get used to).

**Reply by pberna67** [reply]
Dear sfromis
Thanks for the answer
Can you tell me more how to include the file system in the build and the SD card object?
Thanks

**Reply by sfromis** [reply]
With ESP32, I'm using a standard precompiled build, and gpio configuration with pins SPI MOSI/MISO/CLK and SDCard CS.

**Reply by pberna67** [reply]
Where can I find the doc of the SDCard API, or better an fw example which describe how to access to a file system memory card ?

**Reply by sfromis** [reply]
The Berry `file` class methods (and `open` function to create the instance) are described in the "Short Manual" linked from https://github.com/berry-lang/berry/tree/master#readme
To have the file system refer to SD Card files (instead of ESP32 flash), use path names like `/sd/myfile.txt`

Beware that when coding in Berry embedded in Tasmota, you should not have loops running, but instead schedule timer callbacks, as all Berry code should return quickly.

**Reply by sfromis** [reply]
BTW. If those DS18B20 sensors are the usual ones in a metal tube. the thermal mass is so high that saving data every second means that the time correlation will be way off.

**Reply by stefanbode** [reply]
Yes I use them to control my pool temperature and even with heat past I would say about a minute you should calculate to stabilize on a new temp.
